Tests/MathLib/mechanics_invariants.cpp

```diff
@@ -0,0 +1,286 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+#include <gtest/gtest.h>
+#include <autocheck/autocheck.hpp>
+
+#include "MathLib/mechanics_invariants.h"
+
+#include "AutoCheckTools.h"
+
+using namespace MathLib;
+namespace ac = autocheck;
+
+template <int Size>
+using KelvinVector = Eigen::Matrix<double, Size, 1>;
+
+template <int KelvinVectorSize>
+struct KelvinToTensorSize;
+template <>
+struct KelvinToTensorSize<1>
+{
+    static const int value = 1;
+};
+template <>
+struct KelvinToTensorSize<4>
+{
+    static const int value = 2;
+};
+template <>
+struct KelvinToTensorSize<6>
+{
+    static const int value = 3;
+};
+
+template <int Size>
+Eigen::Matrix<double, KelvinToTensorSize<Size>::value,
+              KelvinToTensorSize<Size>::value>
+kelvinToTensor(KelvinVector<Size> const& v);
```

Comment:
I'd prefer to keep it here, otherwise somebody could start doing funny things like
T2K(super_func(K2T(v))) potentially violating the requirement of the Kelvin mapping, symmetry or what not.